examples/cpp/topic_instances/CLIParser.hpp

```diff
@@ -0,0 +1,739 @@
+        publisher_config pub_config;
+        subscriber_config sub_config;
+    };
+
+    /**
+     * @brief Print usage help message and exit with the given return code
+     *
+     * @param return_code return code to exit with
+     *
+     * @warning This method finishes the execution of the program with the input return code
+     */
+    static void print_help(
+            uint8_t return_code)
+    {
+        std::cout << "Usage: topic_instances <entity> [options]"                                << std::endl;
+        std::cout << ""                                                                         << std::endl;
+        std::cout << "Entities:"                                                                << std::endl;
+        std::cout << "  publisher                       Run a publisher entity"                 << std::endl;
+        std::cout << "  subscriber                      Run a subscriber entity"                << std::endl;
+        std::cout << ""                                                                         << std::endl;
+        std::cout << "Common options:"                                                          << std::endl;
+        std::cout << "  -h, --help                      Print this help message"                << std::endl;
+        std::cout << "  -d <num>, --domain <num>        Domain ID number [0 <= <num> <= 232]"   << std::endl;
+        std::cout << "                                  (Default: 0)"                           << std::endl;
+        std::cout << "  -i <num>, --instances <num>     Number of instances to create "         << std::endl;
+        std::cout << "                                  [0 < <num> <= 10 (N colors)]"           << std::endl;
+        std::cout << "                                  (Default: 4 [4 colors])"                << std::endl;
+        std::cout << "  -n <str>, --name <str>          Shape Topic name. Possible values are:" << std::endl;
+        std::cout << "                                   · Square "                             << std::endl;
+        std::cout << "                                   · Triangle "                           << std::endl;
+        std::cout << "                                   · Circle "                             << std::endl;
+        std::cout << "                                  (Default: Square) "                     << std::endl;
+        std::cout << "  -s <num>, --samples <num>       Number of samples per instance to send" << std::endl;
+        std::cout << "                                  or receive.    [0 <= <num> <= 65535]"   << std::endl;
+        std::cout << "                                  (Default: 0 [unlimited])"               << std::endl;
+        std::cout << "Publisher options:"                                                       << std::endl;
+        std::cout << "  -c <str>, --color <str>         Shape color. Possible values are: "     << std::endl;
```

Comment:
The short argument should always be available if possible. In those which has the same initial letter, the short option is not explicit so it is not added.

Reply:
I agree but since you couldn't add any of the other short argument for me it's better if we remove also this one to be consistent. This is in my opinion so feel free to leave it as it is if you don't agree